onadata/apps/api/tests/viewsets/test_xform_submission_viewset.py

```diff
@@ -1232,6 +1233,55 @@ def test_post_submission_using_project_pk_while_authenticated(self):
                     Instance.objects.filter(xform=self.xform).count(), count + 1
                 )
 
+    def test_post_submission_using_project_pk_exceptions(self):
+        """
+        Test that one is able to submit data using the project
+        submission endpoint built for a particular form through
+        it's primary key while authenticated
+        """
+        with self.assertRaises(NoReverseMatch):
+            _url = reverse("submission", kwargs={"project_id": "mission"})
+        s = self.surveys[0]
+        media_file = "1335783522563.jpg"
+        path = os.path.join(
+            self.main_directory,
+            "fixtures",
+            "transportation",
+            "instances",
+            s,
+            media_file,
+        )
+        with open(path, "rb") as f:
+            f = InMemoryUploadedFile(
+                f, "media_file", media_file, "image/jpg", os.path.getsize(path), None
+            )
+            submission_path = os.path.join(
+                self.main_directory,
+                "fixtures",
+                "transportation",
+                "instances",
+                s,
+                s + ".xml",
+            )
+            with open(submission_path, "rb") as sf:
+                data = {"xml_submission_file": sf, "media_file": f}
+                count = Instance.objects.filter(xform=self.xform).count()
+                request = self.factory.post(
+                    f"/projects/{self.xform.project.pk}hello/submission", data
+                )
+                response = self.view(request)
+                self.assertEqual(response.status_code, 401)
+                auth = DigestAuth("bob", "bobbob")
+                request.META.update(auth(request.META, response))
+                response = self.view(request, project_pk=f"{self.project.pk}hello")
+                self.assertEqual(response.status_code, 400)
```

Comment:
Should we check of the validation mesage?

Reply:
Perhaps. Missed this.

Reply:
Should be addressed by https://github.com/onaio/onadata/pull/2742